packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.service.js

```diff
@@ -261,4 +262,43 @@ export default class HostingCdnSharedService {
       .get(`/me/order/${orderId}/status`)
       .then(({ data }) => data);
   }
+
+  /**
+   * Call this to know if upgrade is feasible
+   * @param {string} serviceName: product name
+   * @param {Service} HostingCdnOrderService: hosting order service
+   * @returns {*}
+   */
+  simulateUpgrade(serviceName, HostingCdnOrderService) {
```

Comment:
why are you passing HostingCdnOrderService as a parameter ?
you can inject it with ngInject in the controller of this service no?

Reply:
Because the `HostingCdnOrderService` instantiate `HostingCdnOrderService`.
So i cant instantiate the `HostingCdnOrderService` else i will got a circular cyclic error.

Reply:
I think you want to say that `HostingCdnOrderService` requires `HostingCdnSharedService` ? 
But it's a bit confusing to inject `HostingCdnOrderService` in `HostingCdnSharedService` that is also a dependency of `HostingCdnOrderService`. 
I mean it's not a "circular cyclic error" but a "circular cyclic pattern".

Reply:
Do we have another way to avoid the `Circular dependency found` ?

Reply:
fixed